src/datastore/datastore.rs

```diff
@@ -269,6 +288,10 @@ impl DatastoreInstance {
                 hostname: row.get(4)?,
                 created: row.get(5)?,
                 data: serde_json::from_str(&data_str).unwrap(),
+                metadata: BucketMetadata {
+                    first_starttime: opt_first_starttime,
+                    last_endtime: opt_last_endtime,
```

Comment:
I don't really like the names `first_starttime` and `last_endtime`. Perhaps just `begins` and `ends`? Or a tuple (or ISO8601 formatted interval string) `interval` or `range`?